src/common/config/env.validation.ts

```diff
@@ -98,78 +106,102 @@ export class EnvironmentVariables {
   @Transform(({ value }) => value || LogFormat.json)
   LOG_FORMAT: LogFormat = LogFormat.json;
 
+  @IsNotEmpty()
   @IsArray()
   @ArrayMinSize(1)
-  @Transform(({ value }) => value.split(',').map((url) => url.replace(/\/$/, '')))
+  @IsUrl({
```

Comment:
Will this check run after Transform ?

Reply:
The short answer is yes. The `@Transform()` decorator is always executed first. Then all other validation decorators are executed.

It works this way.

1. If the `PROVIDERS_URLS` is not presented in the `.env` file, the function in the `@Transform()` decorator is not called. As the default value is not assigned to the `PROVIDERS_URLS` variable, the `@IsNotEmpty()` and other decorators will return the appropriate error.

2. If there is a value for the `PROVIDERS_URLS` in the `.env.` file (including the empty value `PROVIDERS_URLS=`), the function in the `@Transform()` decorator will be called first, and then all other validation decorators will be called.